### PR TITLE
Update SSE documentation and remove the feature flag

### PR DESCRIPTION
**Pull request by vladem**
## Description of change

Update documentation to include new SSE options, remove the feature flag.

Relevant issues: https://github.com/awslabs/mountpoint-s3/issues/534

## Does this change impact existing behavior?

This is not a breaking change.

---

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license and I agree to the terms of the [Developer Certificate of Origin (DCO)](https://developercertificate.org/).

### CODE REVIEW

File: doc/CONFIGURATION.md [hunk 1]
```diff
@@ -158,7 +158,9 @@ If necessary, you can use the `--endpoint-url` command-line argument to fully ov
...
+
+New objects can be encrypted using different SSE settings than the bucket's default. To use a different setting, the argument `--sse <aws:kms|aws:kms:dsse|AES256>` should be used to specify the SSE encryption type. When either `aws:kms` or `aws:kms:dsse` is used as a type, `--sse-kms-key-id <KEY_ARN>` may be used to optionally specify a KMS key ID. For these encryption types, when the `--sse` argument is used without providing a KMS key ID, S3 will use an [AWS managed KMS key](https://docs.aws.amazon.com/kms/latest/developerguide/concepts.html#key-mgmt), which is created automatically. Please note that these command line arguments only configure server-side encryption for *new* objects created with Mountpoint, all of the *existing* objects will remain encrypted with settings specified at the time of their creation.
```
Comment (dannycjones) [inline doc/CONFIGURATION.md:163]:
```suggestion
New objects can be uploaded using different server-side encryption (SSE) settings than the bucket's default. The CLI argument `--sse <aws:kms|aws:kms:dsse|AES256>` can be used to specify a different SSE encryption type. When either `aws:kms` or `aws:kms:dsse` is used as a type, `--sse-kms-key-id <KEY_ARN>` may be used to optionally specify a KMS key ID. When a KMS key ID is not specified, S3 will use an [AWS managed KMS key](https://docs.aws.amazon.com/kms/latest/developerguide/concepts.html#key-mgmt), which is created automatically. Please note that these command-line arguments only configure server-side encryption for *new* objects created with Mountpoint, all *existing* objects will remain unchanged.
```

File: mountpoint-s3/Cargo.toml [hunk 2]
```diff
@@ -77,7 +77,6 @@ built = { version = "0.7.1", features = ["git2"] }
 [features]
 # Unreleased feature flags
 negative_cache = []
-sse_kms = []
```
Comment (dannycjones) [inline mountpoint-s3/Cargo.toml:80]: Perhaps an alternative approach in future is that we leave the feature flag in but remove all conditions throughout the code, then follow-up to remove it.

**Review by dannycjones — CHANGES_REQUESTED** [review]
Small tweaks to the doc changes. Otherwise, LGTM.

**Comment by dannycjones** [comment]
I updated the title just so the eventual squash commit is consistent with previous style.

**Review by dannycjones — APPROVED** [review]
LGTM.

File: .github/workflows/integration.yml [hunk 3]
```diff
@@ -30,7 +30,7 @@ env:
   S3_OLAP_ARN: ${{ vars.S3_OLAP_ARN }}
   S3_MRAP_ARN: ${{ vars.S3_MRAP_ARN }}
   KMS_TEST_KEY_ID: ${{ vars.KMS_TEST_KEY_ID }}
-  RUST_FEATURES: fuse_tests,s3_tests,fips_tests
+  RUST_FEATURES: fuse_tests,s3_tests,fips_tests,sse_kms
```
Comment (dannycjones) [inline .github/workflows/integration.yml:33]: Why add it back?

Comment (vladem) [reply .github/workflows/integration.yml]: something went wrong with rebasing, sorry

**Review by dannycjones — APPROVED** [review]
LGTM!

**Pull request merged**